internal/idle/idle_e2e_test.go

```diff
@@ -215,15 +227,16 @@ func (s) TestChannelIdleness_Enabled_OngoingCall(t *testing.T) {
 	testutils.AwaitState(ctx, t, cc, connectivity.Ready)
 
 	// Spawn a goroutine which checks expected state transitions and idleness
-	// channelz trace events. It eventually closes `blockCh`, thereby unblocking
-	// the server RPC handler and the unary call below.
+	// channelz trace events.
 	errCh := make(chan error, 1)
 	go func() {
-		defer close(blockCh)
 		// Verify that the ClientConn stays in READY.
 		sCtx, sCancel := context.WithTimeout(ctx, 3*defaultTestShortIdleTimeout)
 		defer sCancel()
-		testutils.AwaitNoStateChange(sCtx, t, cc, connectivity.Ready)
+		if cc.WaitForStateChange(sCtx, connectivity.Ready) {
+			errCh <- fmt.Errorf("State changed from %q to %q when no state change was expected", connectivity.Ready, cc.GetState())
```

Comment:
If you're expecting no change here, then the `if` condition doesn't express that.  I think this is just wrong, in fact. The "If" is waiting for the state to _become_ READY. So if the state changes to anything else, it will not be caught.

Reply:
This is exactly what `testutils.AwaitNoStateChange` does.

> The `if` is waiting for the state to become READY

```
// WaitForStateChange waits until the connectivity.State of ClientConn changes from sourceState or
// ctx expires. A true value is returned in former case and false in latter.
```

No, it is waiting for the state to change out of READY or the context to expire.

Reply:
I've not used this tooltip feature very much at all so far. Won't be using it in the future :)